onnxscript/rewriter/pattern.py

```diff
@@ -1137,11 +1149,13 @@ def rewrite(cls, op, *_) -> Any:
         raise NotImplementedError("Method 'rewrite' must be overwritten.")
 
     @classmethod
-    def check(cls, context, *_) -> bool:
+    def check(cls, context, *_, **__) -> bool:
         return True
 
 
-def make_rewrite_rule_from_class(rule_class: type | RewriteRuleAsClass) -> RewriteRule:
+def make_rewrite_rule_from_class(
+    rule_class: type | RewriteRuleAsClass, generic: bool = False
```

Comment:
Wanted to understand: why do we need this option to distinguish between them? I would think we have already merged the two algorithms and the algo choice was internal.

Reply:
Ideally this shouldn't be a user choice, as seen here.

Reply:
Two algorithms not doing exactly the same thing. The can switch if one is failing.